docs/guide/extensibility/completions.md

```diff
@@ -186,7 +186,34 @@ even if the prefix only partially matches
 the snippets' tab triggers.
 
 ::: tip Note
-Completions sourced from words in the buffer 
-can be suppressed explicitly 
+Completions sourced from words in the buffer
+can be suppressed explicitly
 from an `on_query_completions` event hook.
 :::
+
+
+## Auto Completion Triggers and Selector
+
+Sublime Text provides two settings
+for users and package authors
+to tweak their auto completion behavior.
+
+- `auto_complete_selector` expects a scope selector
+  that when matched causes the auto completion window
+  to open for any non-punctuation character,
+  i.e. for identifiers.
+  The default configuration disables this
+  for syntaxes with a focus on plain text, comments and strings.
+
+  Before 4070,
+  the selector was applied to the position *after* the just-typed character.
+
+- `auto_complete_triggers` expects a list of mappings
+  that may specify a character that should trigger the auto completion window
+  and/or a scope selector.
+  If both are provided,
+  both are required to match.
+
+Package authors may be interested
+in configuring triggers and/or the selector
+in a syntax-specific settings file.

```

Comment:
Is it wise to suggest that plugins can modify the global `auto_complete_selector`? I'm fairly confident in saying a plugin can accomplish any sort of optimal behavior by adding a selector via `auto_complete_triggers`.

Reply:
`auto_complete_triggers` is a global setting as well, so I'm not sure how suggesting that over the other makes a difference. Regardless, I think you might have missed the mention of "syntax-specific" settings files there? Because that's really important, we could probably emphasize that.

Reply:
If a plugin replaces auto_complete_selector then it is replaced. While auto_complete_triggers can be added (in some `__init__`) and removed (in some `__del__`). IMO, it's better to slightly adjust the combined selector of auto_complete_selector and the OR of all selectors of auto_complete_triggers by adding a selector to the list of auto_complete_triggers and appropriately removing it when the plugin is disabled.

Reply:
`auto_complete_triggers`, even as a list, needs to be replaced by a plugin just like `auto_complete_selector`. To restore it to its previous value, the plugin needs to either cache the previous value or erase its view-specific override. I don't think the exact practices matter in this case, since both settings have specific use cases. PackageDev makes use of both, for example, to suggest key completions.